src/JanusGraph.Net/Geoshapes/Point.cs

```diff
@@ -0,0 +1,79 @@
+#region License
+
+/*
+ * Copyright 2018 JanusGraph.Net Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#endregion
+
+using System;
+
+namespace JanusGraph.Net.Geoshapes
+{
+    /// <summary>
+    ///     A single point representation. A point is identified by its coordinate on the earth sphere using the spherical
+    ///     system of latitudes and longitudes.
+    /// </summary>
+    public class Point : IEquatable<Point>
+    {
+        internal Point(double latitude, double longitude)
+        {
+            Latitude = latitude;
+            Longitude = longitude;
+        }
+
+        /// <summary>
+        ///     Gets the latitude of this point.
+        /// </summary>
+        public double Latitude { get; }
+
+        /// <summary>
+        ///     Gets the longitude of this point.
+        /// </summary>
+        public double Longitude { get; }
+
+        /// <inheritdoc />
+        public bool Equals(Point other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return Latitude.Equals(other.Latitude) && Longitude.Equals(other.Longitude);
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+            return Equals((Point) obj);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                return (Latitude.GetHashCode() * 397) ^ Longitude.GetHashCode();
+            }
+        }
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            return $"POINT ({Longitude} {Latitude})";
```

Comment:
add a comment describing why the order is this way

Reply:
I think this would be cleaner with directions for example `POINT (10.12345 N -12.3434 E)

Reply:
@pluradj: What is the reason for the ordering? I only copied it from JanusGraph and only found this sentence in the docs:
> Note that unlike the JanusGraph API GeoJSON specifies coordinates as lng lat.

@farodin91: I only know it in this form if the orientation is added: `51° 28′ 55″ N, 7° 12′ 57″ E`. The current implementation corresponds to the Java implementation. So, we stay consistent across languages.

Reply:
I am not aware of specific implementation, but it seems that there is a bug in JanusGraph coordinates. I see that in some places we use `lat lng` and in other places `lng lat`. As a result, there is a confusion which leads to new bugs. Right now I've found a bug in ElasticSearch index implementation https://github.com/JanusGraph/janusgraph/issues/1291 . I didn't check Solr and Lucene but there is a possibility that this bug is presented there also.

P.S. I've never seen `lng lat` order before in other systems. I think that we should stick to one style (I prefer `lat lng`).

Reply:
JanusGraph expects the form lat lng, but uses [spatial4j](https://github.com/locationtech/spatial4j) internally which seems to use lng lat. I mirrored the Java API here to stay at least consistent with that.
I agree that it's confusing like it is right know, but I think that we should at least be consistent across languages. So, if we want to change the order of coordinates, then we should first do that in JanusGraph itself before we change it in a library like JanusGraph.Net in my opinion.